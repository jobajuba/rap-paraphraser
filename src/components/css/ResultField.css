.result {
    border: 1px solid black;
    height: 200px;
    overflow-y: auto;
}

.result.loading {
     /*background: url("../../assets/img/loading.gif") no-repeat center;*/
 }
