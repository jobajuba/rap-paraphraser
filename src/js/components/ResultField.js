import React from 'react';
import classNames from 'classnames';

export const ResultField = (props) => {
    const {text = '', baseClass = '', isLoading = false} = props;
    const classes = classNames({
        [baseClass]: true,
        'loading': isLoading,
    });
    return (
        <div id="result" className={classes}>{text}</div>
    );
};
