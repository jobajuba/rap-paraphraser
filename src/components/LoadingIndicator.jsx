import React from 'react';
import './css/LoadingIndicator.css';

export const LoadingIndicator = () => {
    return (
        <div className="loader">
            <svg className="circular-loader" viewBox="25 25 50 50">
                <circle
                    className="loader-path"
                    cx="50"
                    cy="50"
                    r="20"
                    fill="none"
                    stroke="#70c542"
                    strokeWidth="2"
                />
            </svg>
        </div>
    );
};
