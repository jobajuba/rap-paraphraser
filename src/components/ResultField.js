import React from 'react';
import classNames from 'classnames';
import './css/ResultField.css';

export const ResultField = (props) => {
    const {text = '', baseClass = '', isLoading = false} = props;
    const classes = classNames({
        [baseClass]: true,
        'result': true,
        'loading': isLoading,
    });
    return (
        <div className={classes}>{text}</div>
    );
};
