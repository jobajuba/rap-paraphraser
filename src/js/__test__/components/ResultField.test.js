import React from 'react';
import renderer from 'react-test-renderer';
import {describe, expect, test} from '@jest/globals';
import {ResultField} from '../../components/ResultField';

describe('Test of ResultField component', () => {

    test('ResultField component render without props', () => {
        const component = renderer.create(
            <ResultField/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ResultField component render with text', () => {
        const component = renderer.create(
            <ResultField text="testText"/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ResultField component render with class', () => {
        const component = renderer.create(
            <ResultField baseClass="testClass"/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ResultField component render with class and loading', () => {
        const component = renderer.create(
            <ResultField baseClass="testClass" isLoading={true}/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ResultField component render with class, text and loading', () => {
        const component = renderer.create(
            <ResultField baseClass="testClass" text="testText" isLoading={true}/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

});
