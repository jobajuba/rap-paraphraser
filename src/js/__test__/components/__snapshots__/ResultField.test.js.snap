// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of ResultField component ResultField component render with class 1`] = `
<div
  className="testClass"
  id="result"
>
  
</div>
`;

exports[`Test of ResultField component ResultField component render with class and loading 1`] = `
<div
  className="testClass loading"
  id="result"
>
  
</div>
`;

exports[`Test of ResultField component ResultField component render with class, text and loading 1`] = `
<div
  className="testClass loading"
  id="result"
>
  testText
</div>
`;

exports[`Test of ResultField component ResultField component render with text 1`] = `
<div
  className=""
  id="result"
>
  testText
</div>
`;

exports[`Test of ResultField component ResultField component render without props 1`] = `
<div
  className=""
  id="result"
>
  
</div>
`;
